oak_runtime/src/node/http/server.rs

```diff
@@ -358,6 +362,21 @@ impl HttpRequestHandler {
     }
 }
 
+/// Check if the request contains a `TRANSFER_ENCODING` header, and reject the request in that case
+/// by returning an error.
+fn validate_request(req: &HttpRequest) -> anyhow::Result<()> {
```

Comment:
How about making this accept an `HttpRequest` (by value) and also return a `Result<HttpRequest>`, so that there is no risk that the caller misses it? I think the `Result<()>` return type is always a bit too easy to miss / misuse.

Reply:
+1. I would even consider doing this validation in `to_oak_http_request` seeing that it is already returning a result. This would also mean that you can use the http library's headermap to check for existence of the header before mapping it to the proto headermap.

Reply:
I like it. Thanks. 
I also changed it to take the original request as input. But I am keeping it separate from `to_oak_http_request` to be explicit that a validation step is also happening.